src/src/PluginStructs/P121_data_struct.cpp

```diff
@@ -0,0 +1,44 @@
+#include "../PluginStructs/P121_data_struct.h"
+
+#ifdef USES_P121
+
+// Needed also here for PlatformIO's library finder as the .h file
+// is in a directory which is excluded in the src_filter
+#include <Adafruit_Sensor.h>
+#include <Adafruit_HMC5883_U.h>
+
+bool P121_data_struct::begin()
+{
+  if (!initialized)
+  {
+    initialized = mag.begin();
+
+    if (initialized)
+    {
+      // Set up oversampling and filter initialization
+      sensor_t sensor;
+      mag.getSensor(&sensor);
+      #ifndef BUILD_NO_DEBUG
+      addLog(LOG_LEVEL_DEBUG, F("------------------------------------"));
```

Comment:
Can you also wrap it in `if (logLevelActiveFor(LOG_LEVEL_DEBUG)) { ... }` ?

Reply:
Ok